qiita_db/test/test_data.py

```diff
@@ -478,6 +478,38 @@ def test_create_data_type_only(self):
         # preprocessed_data_id, filepath_id
         self.assertEqual(obs, [[3, obs_id - 1], [3, obs_id]])
 
+    def test_delete(self):
+        """Correctly deletes a preprocessed data"""
+        # testing regular delete
+        ppd = PreprocessedData.create(
+            self.study, self.params_table,
+            self.params_id, self.filepaths, prep_template=self.prep_template,
+            ebi_submission_accession=self.ebi_submission_accession,
+            ebi_study_accession=self.ebi_study_accession)
+        PreprocessedData.delete(ppd.id)
```

Comment:
@josenavas is right, the FK restriction and the usage of queues should keep this sane, @squirrelo can you still see a scenario where we would not be safe?

However I don't think a check for "trying to delete" again should be the way to verify things work, the test below should check it can't instantiate the object. And for completeness there should be another test trying to delete an object that doesn't exist (maybe this same one).